crates/bevy_asset/src/lib.rs

```diff
@@ -231,13 +340,24 @@ impl Plugin for AssetPlugin {
     }
 }
 
+/// Declares that this type is an asset,
+/// which can be loaded and managed by the [`AssetServer`] and stored in [`Assets`] collections.
+///
+/// Generally, assets are large, complex, and/or expensive to load from disk, and are often authored by artists or designers.
```

Comment:
```suggestion
/// Generally, assets are large, complex, and/or expensive to load from disk, and are often authored by artists or designers.
/// Examples include textures, sounds, 3D models, etc.
```

Reply:
The other thing you might want to mention is hot reloading "data", e.g. how much damage an enemy does or what items they drop or their spawn rate or whatever.

Reply:
Actually, I'm not seeing any mention of hot reloading at all in these docs, seems like a missing piece :)

Reply:
So when I was reading through with #3130 in mind, the biggest question remaining for me is timing-sensitive code, i.e. should we be calling `asset_server.load` in an `Update` system, because my instinct is "no" but I don't really have anything to back that up except the number of times updates get called in a second! So, something around general runtime changes: we're after the initial load, do we:
- place an asset load request in a queue to be processed
- load the asset in update systems, because this is considered safe (?)
- something else
And touch on whether any of these mechanisms should be state-controlled. That's the missing piece of the puzzle for me, I guess as a _junior apprentice_ :laughing:

Reply:
Yeah, IMO this deserves a full example to demonstrate the patterns here.

Reply:
If only someone was working on such a thing! Oh, wait...